File: bridge/typescript/email-message.ts
import {Message} from "./message";
import {IMessageSender} from "./message-sender.interface";

// UML diyagramındaki RefinedAbstraction kısmına denk gelmektedir.
// Abstraction (Message) sınıfından türer.
export class EmailMessage extends Message {
  constructor(protected readonly messageSender: IMessageSender) {
    super(messageSender);
  }

  async send(): Promise<void> {
    await this.messageSender.sendMessage(this);
  }
}

File: bridge/go/email-message.go
package main

type emailMessage struct {
	messageSender MessageSender
}

func NewEmailMessage(messageSender MessageSender) *emailMessage {
	return &emailMessage{
		messageSender: messageSender,
	}
}

func (m *emailMessage) Send() {
	m.messageSender.sendMessage(m)
}
